src/python/gudhi/simplex_tree.pyx

```diff
@@ -214,15 +215,14 @@ cdef class SimplexTree:
         :returns:  The simplices sorted by increasing filtration values.
         :rtype:  list of tuples(simplex, filtration)
         """
-        cdef vector[pair[vector[int], double]] filtration \
-            = self.get_ptr().get_filtration()
-        ct = []
-        for filtered_complex in filtration:
-            v = []
-            for vertex in filtered_complex.first:
-                v.append(vertex)
-            ct.append((v, filtered_complex.second))
-        return ct
+        cdef vector[Simplex_tree_simplex_handle].const_iterator it = self.get_ptr().get_filtration_iterator_begin()
+        cdef vector[Simplex_tree_simplex_handle].const_iterator end = self.get_ptr().get_filtration_iterator_end()
+
+        while True:
+            yield(self.get_ptr().get_simplex_filtration(dereference(it)))
+            preincrement(it)
+            if it == end:
+                raise StopIteration
```

Comment:
You don't have to raise the exception yourself, `break` to end the loop is enough (getting to the end of the function is how you finish iterating, with yield generators). So probably `while it != end:` instead?